File: Inventarios.Api/Inventarios.api/Controllers/AuthController.cs
using Inventarios.api.Application.UseCases;
using Inventarios.api.Domain.Dtos.Requests;
using Inventarios.api.Domain.Dtos.Responses;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Inventarios.api.Controllers;

[ApiController]
[Route("api/[controller]")]
[Produces("application/json")]
public sealed class AuthController : ControllerBase
{
    private readonly LoginUserUseCase _loginUseCase;

    public AuthController(LoginUserUseCase loginUseCase)
    {
        _loginUseCase = loginUseCase;
    }

    /// <summary>Autentica un usuario y retorna un token JWT.</summary>
    [HttpPost("login")]
    [AllowAnonymous]
    [ProducesResponseType<ApiResponse<AuthResponse>>(StatusCodes.Status200OK)]
    [ProducesResponseType<ApiResponse<object>>(StatusCodes.Status401Unauthorized)]
    public async Task<IActionResult> Login([FromBody] LoginRequest request, CancellationToken ct)
    {
        var result = await _loginUseCase.ExecuteAsync(request, ct);

        if (result.IsFailure)
            return Unauthorized(ApiResponse<object>.Fail(result.Error!));

        return Ok(ApiResponse<AuthResponse>.Ok(result.Value!, "Autenticación exitosa."));
    }
}

File: Inventarios.Api.Nest/productos.api/src/presentation/controllers/auth.controller.ts
import {
  Controller, Post, Body, HttpCode, HttpStatus
} from '@nestjs/common';
import { LoginUseCase } from '../../application/use-cases/auth/login.use-case';
import { LoginRequest } from '../../domain/dtos/requests/login.request';
import { ApiResponse } from '../../domain/dtos/responses/api.response';

@Controller('api/auth')
export class AuthController {
  constructor(private readonly loginUseCase: LoginUseCase) {}

  @Post('login')
  @HttpCode(HttpStatus.OK)
  async login(@Body() request: LoginRequest) {
    const result = await this.loginUseCase.execute(request);

    if (result.isFailure)
      return ApiResponse.fail(result.error ?? 'Error al autenticar.');

    return ApiResponse.ok(result.value, 'Autenticación exitosa.');
  }
}